ome_zarr/format.py

```diff
@@ -183,11 +183,15 @@ def init_store(self, path: str, mode: str = "r") -> FSStore:
         TODO: could also check dimension_separator
         """
 
-        kwargs = {
-            "dimension_separator": "/",
+        kwargs: Dict[str, Any] = {
             "normalize_keys": False,
         }
 
+        # if creating a store, use / by default. If reading, use the
+        # dimension_separator specified in the .zarray
+        if mode == "w":
```

Comment:
There certainly _were_ datasets that had no `dimension_separator` in the .zarray and therefore this would make them unreadable. I'd suggest we do this with the bump of the spec that says we no longer support data written before dimension_separator was introduced.

Reply:
If datasets have no `dimension_separator` then they will be using `.` which is the zarr default, so that will be used and we will be able to read them.
Whereas, without this PR they won't be readable.
However, the spec for v0.2 - 0.4 says they should use `/`, so this PR is about adding support for "invalid" data, but we have seen plenty of examples where that exists.

Reply:
> However, the spec for v0.2 - 0.4 says they should use /, so this PR is about adding support for "invalid" data, but we have seen plenty of examples where that exists.

Understood but there are (at least) "two" types of invalid data.

Reply:
Ah - OK, I understand the possible existence of e.g. v0.2 data that uses `/` dimension_separator but doesn't specify the `dimension_separator` in the `.zarray`.
But I'm not so sure about how the next release of NGFF spec (v0.5) should affect the support for `ome-zarr-py` to read that data, since the spec doesn't specify how libraries should support old versions of data?

Reply:
The plan I've been working towards is dropping any requirement on the dimension separator in 0.5 other than to say the Zarr implementation used is responsible for _either_ using the default of `.` or of setting `dimension_separator` correctly. i.e. not our business. But the current specs <0.5 explicitly state the dimension separator to use and therefore I would suggest not changing those semantics.

cc: @normanrz @kevinyamauchi